cpu/sam0_common/include/periph_cpu_common.h

```diff
@@ -670,6 +670,218 @@ typedef struct {
+ */
+#if defined(CPU_FAM_SAML21)
+#define DMA_DESCRIPTOR_IN_LPSRAM
+#endif
+
+#ifdef DMA_DESCRIPTOR_IN_LPSRAM
+#define DMA_DESCRIPTOR_ATTRS    __attribute__((section(".backup.bss")))
+#else
+#define DMA_DESCRIPTOR_ATTRS
+#endif
+
+/**
+ * @brief DMA channel type
+ */
+typedef unsigned dma_t;
+
+/**
+ * @brief Available DMA address increment modes
+ */
+typedef enum {
+    DMA_INCR_NONE   = 0,    /**< Don't increment any addresses after a beat */
+    DMA_INCR_SRC    = 1,    /**< Increment the source address after a beat */
+    DMA_INCR_DEST   = 2,    /**< Increment destination address after a beat */
+    DMA_INCR_BOTH   = 3,    /**< Increment both addresses after a beat */
+} dma_incr_t;
+
+/**
+ * @brief   Initialize DMA
+ */
+void dma_init(void);
+
+/**
+ * @brief Acquire a DMA channel.
+ *
+ * A free DMA channel is marked as allocated and a reference is returned.
+ * DMA channels can be acquired for long periods of time, e.g. from the start to
+ * end of a number of transfers or directly at boot and never released.
+ *
+ * @returns     A reference to the DMA channel
+ * @returns     UINT8_MAX when no DMA channel is available
+ */
+dma_t dma_acquire_channel(void);
+
+/**
+ * @brief   Release a previously acquired DMA channel
+ *
+ * @param   dma     DMA channel to release
+ */
+void dma_release_channel(dma_t dma);
+
+/**
+ * @brief   Initialize a previously allocated DMA channel with one-time settings
+ *
+ * @param   dma     DMA channel reference
+ * @param   trigger Trigger to use for this DMA channel
+ * @param   prio    Channel priority
+ * @param   irq     Whether to enable the interrupt handler for this channel
+ */
+void dma_setup(dma_t dma, unsigned trigger, uint8_t prio, bool irq);
+
+/**
+ * @brief   Prepare the DMA channel for an individual transfer.
+ *
+ * @param   dma     DMA channel reference
+ * @param   width   Transfer beat size to use
+ * @param   src     Source address for the transfer
+ * @param   dst     Destination address for the transfer
+ * @param   len     Number of beats to transfer
+ * @param   incr    Which of the addresses to increment after a beat
+ */
+void dma_prepare(dma_t dma, uint8_t width, void *src, void *dst, size_t len,
+                 dma_incr_t incr);
+
+/**
+ * @brief   Prepare a transfer without modifying the destination address
+ *          settings.
+ *
+ * Can be used when repeatedly using a dma channel to transfer to the same
+ * peripheral address, leaving the destination address and related settings
+ * untouched
+ *
+ * @note This only touches the source address, length and source increment
+ * settings. Be sure to initialize the full descriptor beforehand with
+ * @ref dma_prepare
+ *
+ * @param   dma     DMA channel reference
+ * @param   src     Source address for the transfer
+ * @param   len     Number of beats to transfer
+ * @param   incr    Whether to increment the source address after a beat
+ */
+void dma_prepare_src(dma_t dma, void *src, size_t len, bool incr);
```

Comment:
Yea I'm late to the game, but just noticed this when reading the SPI implementation.

Why not
```suggestion
void dma_prepare_src(dma_t dma, const void *src, size_t len, bool incr);
```
?

Reply:
Yup, that should be `const` indeed, same for the other occurrences of `src`